doc/source/user/basics.types.rst

```diff
@@ -12,87 +12,124 @@ Array types and conversions between types
+      - Boolean (True or False) stored as a byte.
 
-    * - `numpy.byte`
+    * - `numpy.int8`
+      - `numpy.byte`
       - ``signed char``
-      - Platform-defined
+      - Platform-defined integer type with 8 bits.
 
-    * - `numpy.ubyte`
+    * - `numpy.uint8`
+      - `numpy.ubyte`
       - ``unsigned char``
-      - Platform-defined
+      - Platform-defined integer type with 8 bits without sign.
 
-    * - `numpy.short`
+    * - `numpy.int16`
+      - `numpy.short`
       - ``short``
-      - Platform-defined
+      - Platform-defined integer type with 16 bits.
 
-    * - `numpy.ushort`
+    * - `numpy.uint16`
+      - `numpy.ushort`
       - ``unsigned short``
-      - Platform-defined
+      - Platform-defined integer type with 16 bits without sign.
 
-    * - `numpy.intc`
+    * - `numpy.int32`
+      - `numpy.intc`
       - ``int``
-      - Platform-defined
+      - Platform-defined integer type with 32 bits.
 
-    * - `numpy.uintc`
+    * - `numpy.uint32`
+      - `numpy.uintc`
       - ``unsigned int``
-      - Platform-defined
+      - Platform-defined integer type with 32 bits without sign.
+
+    * - `numpy.intp`
+      - N/A
+      - ``intptr_t``
+      - Platform-defined integer type capable of holding a pointer.
 
-    * - `numpy.int_`
+    * - `numpy.uintp`
+      - N/A
+      - ``uintptr_t``
+      - Platform-defined integer type capable of holding a pointer without sign.
+
+    * - `numpy.int32` or `numpy.int64`
+      - `numpy.long`
       - ``long``
-      - Platform-defined
+      - Platform-defined integer type with at least 32 bits.
 
-    * - `numpy.uint`
+    * - `numpy.uint32` or `numpy.uint64`
+      - `numpy.ulong`
       - ``unsigned long``
-      - Platform-defined
+      - Platform-defined integer type with at least 32 bits without sign.
 
-    * - `numpy.longlong`
+    * - N/A
+      - `numpy.longlong`
       - ``long long``
-      - Platform-defined
+      - Platform-defined integer type with at least 64 bits.
 
-    * - `numpy.ulonglong`
+    * - N/A
+      - `numpy.ulonglong`
       - ``unsigned long long``
-      - Platform-defined
+      - Platform-defined integer type with at least 64 bits without sign.
 
-    * - `numpy.half` / `numpy.float16`
-      -
+    * - `numpy.float16`
+      - `numpy.half`
+      - N/A
       - Half precision float:
-        sign bit, 5 bits exponent, 10 bits mantissa
+        sign bit, 5 bits exponent, 10 bits mantissa.
 
-    * - `numpy.single`
+    * - `numpy.float32`
+      - `numpy.single`
       - ``float``
       - Platform-defined single precision float:
-        typically sign bit, 8 bits exponent, 23 bits mantissa
+        typically sign bit, 8 bits exponent, 23 bits mantissa.
 
-    * - `numpy.double`
+    * - `numpy.float64`
+      - `numpy.double`
       - ``double``
       - Platform-defined double precision float:
         typically sign bit, 11 bits exponent, 52 bits mantissa.
 
-    * - `numpy.longdouble`
+    * - `numpy.float96` or `numpy.float128`
```

Comment:
`float96` doesn't get rendered with a cross-reference in the built version like the other types. You either need to add an API reference entry for `float96` like the other types (apparently it's missing?) or if you don't want to mess with that just use double backticks.

Reply:
I think it's missing because PR docs are built on machine where `float96` doesn't exist (there's `float128` instead). Let me add double backticks.

Reply:
Ah that makes sense, thanks!